docs/plugins/esm-external-require.md

```diff
@@ -63,13 +41,34 @@ export default defineConfig({
 
 Type: `(string | RegExp)[]`
 
-Defines which dependencies should be treated as external and have their `require()` calls converted to imports.
+Defines which dependencies should be treated as external. When the output format is ESM, their `require()` calls will be converted to `import` statements. For non-ESM output formats, the dependencies will be marked as external but the `require()` calls will remain unchanged.
 
 ## Limitations
 
-The plugin only activates when:
+Since this plugin changes `require()` calls to `import` statements, there are some semantic differences after bundling:
 
-- The output format is ESM
```

Comment:
This was confusing to me. I thought the whole plugin won't run, but actually it _does_ run. It just doesn't do the conversion.